src/ghci/mod.rs

```diff
@@ -551,6 +554,11 @@ impl Ghci {
     /// Optionally returns a compilation result.
     #[instrument(skip(self), level = "debug")]
     async fn add_module(&mut self, path: &NormalPath) -> miette::Result<Option<CompilationResult>> {
+        if self.targets.contains_source_path(path.absolute())? {
+            tracing::debug!(%path, "Skipping `:add`ing already-loaded path");
+            return Ok(None);
+        }
```

Comment:
This is _also_ sufficient to fix the bug. I'm adding it here in case I make this programming error in the future (calling `Ghci::add` on a file that's already part of the target set).